exec/java-exec/src/main/java/org/apache/drill/exec/expr/EvaluationVisitor.java

```diff
@@ -479,6 +479,14 @@ private HoldingContainer visitValueVectorReadExpression(ValueVectorReadExpressio
 
       if (!hasReadPath && !complex) {
         JBlock eval = new JBlock();
+
+        if (repeated) {
```

Comment:
Was it the bug?
Am I right, we just didn't reach this part of code with other functions with `Repeated` DataMode?

Reply:
It was kind of a bug. Yes, this code wasn't reached by other functions (at least those I am aware of).